optuna/storages/_in_memory.py

```diff
@@ -166,7 +165,7 @@ def _build_study_summary(self, study_id: int, include_best_trial: bool = True) -
             n_trials=len(study.trials),
             datetime_start=min(
                 [
-                    cast(datetime, trial.datetime_start)
+                    trial.datetime_start
```

Comment:
[Notes] `FrozenTrial.datetime_start` is `Optional[datetime.datetime]` and `FrozenTrial.datetime_start` is not `None` here.

https://github.com/optuna/optuna/blob/71d3a799b0c13b62c8b71688facffce667f2b389/optuna/trial/_frozen.py#L131